keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt

```diff
@@ -297,6 +297,11 @@ abstract class ArtifactRepositoryTests<T : ArtifactRepository> : JUnit5Minutests
           test("querying with only release returns correct versions") {
             expectThat(subject.versions(artifact2)).containsExactly(version5, version4)
           }
+
+          test("querying for limit returns limit") {
+            val artifactWithAll = artifact1.copy(statuses = emptySet())
+            expectThat(subject.versions(artifactWithAll, 2)).containsExactly(version5, version4)
+          }
```

Comment:
Are there more versions than that in the database? It looks like the test right above that does not use the limit also gets 2 results back.

Reply:
this test is calling `persist`, which is storing 5 versions for `artifact1` and we expecting to get only 2 back